pkg/api/v1alpha1/cluster_webhook.go

```diff
@@ -54,21 +55,26 @@ var _ webhook.Validator = &Cluster{}
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type
 func (r *Cluster) ValidateCreate() error {
 	clusterlog.Info("validate create", "name", r.Name)
+	var allErrs []error
+	if err := r.Validate(); err != nil {
+		allErrs = append(allErrs, err)
+	}
+
 	if r.IsReconcilePaused() {
 		clusterlog.Info("cluster is paused, so allowing create", "name", r.Name)
 		return nil
```

Comment:
This is wrong
It means that even if `r.Validate()` fails, we will accept the request

Which makes me think we are missing a set of tests

I would advise to write the logic in a way where we only return at the end of the method. Having errors accumulating in a slice but also intertwined with early returns makes the code more difficult to read.